js/windowMgr.js

```diff
@@ -500,6 +504,77 @@ function doCreateMainWindow(initialUrl, initialBounds, isCustomTitleBar) {
                 });
             });
     });
+    
+    // ELECTRON-323: Handle session permission requests
+    function handlePermissionRequests(webContents) {
+        
+        let session = webContents.session;
+        
+        getConfigField('permissions')
+            .then((permissions) => {
+                
+                if (!permissions) {
+                    log.send(logLevels.ERROR, 'permissions configuration is invalid, so, everything will be true by default!');
+                    return;
+                }
+                
+                session.setPermissionRequestHandler((sessionWebContents, permission, callback) => {
+                    
+                    function handleSessionPermissions(userPermission, message, cb) {
+                        
+                        log.send(logLevels.INFO, 'permission is -> ' + userPermission);
+                        
+                        if (!userPermission) {
+                            let fullMessage = `Your administrator has disabled ${message}. Please contact your admin for help.`;
+                            electron.dialog.showErrorBox('Permission Denied!', fullMessage);
+                        }
+                        
+                        return cb(userPermission);
+                        
+                    }
+                    
+                    let PERMISSION_MEDIA = 'media';
+                    let PERMISSION_LOCATION = 'geolocation';
+                    let PERMISSION_NOTIFICATIONS = 'notifications';
+                    let PERMISSION_MIDI_SYSEX = 'midiSysex';
+                    let PERMISSION_POINTER_LOCK = 'pointerLock';
+                    let PERMISSION_FULL_SCREEN = 'fullscreen';
+                    let PERMISSION_OPEN_EXTERNAL = 'openExternal';
+                    
+                    switch (permission) {
+                        
+                        case PERMISSION_MEDIA:
+                            return handleSessionPermissions(permissions.media, 'sharing your camera, microphone, and speakers', callback);
+                        
+                        case PERMISSION_LOCATION:
+                            return handleSessionPermissions(permissions.geolocation, 'sharing your location', callback);
+                        
+                        case PERMISSION_NOTIFICATIONS:
+                            return handleSessionPermissions(permissions.notifications, 'notifications', callback);
```

Comment:
won't this only stop browser/native notifications?  that is, this won't stop custom notifications we show on windows - correct?   if so, might this be confusing???

Reply:
you are right, I've documented that here (https://perzoinc.atlassian.net/wiki/spaces/DES/pages/320897190/Electron+Wrapper+Permissions) and linked it to the release notes.